docs/advanced/document-store/benchmarking.md

```diff
@@ -0,0 +1,122 @@
+exhaustive search instead. That's why, they give the maximum quality but are the slowest when it comes to search nearest 
+neighbors.
+```
+
+The results were conducted on a 4.5 Ghz AMD Ryzen Threadripper 3960X 24-Core Processor with Python 3.8.5 and using the official docker 
+images of the storage backends. The docker images were allocated 40 GB of RAM.
+
+The benchmarking experiments used the following parameters:
+* Number of indexed Documents: 1M
+* Number of query Documents: 1
+* Embedding dimension: 128
+* Number of Documents (K) to be retrieved for `Find Document by vector`: 10
+
+For the first experiment, we fixed the following HNSW parameters for the Document Stores that support ANN:
+* `ef` (construction): 100
+* `ef` (search): 100
+* `m` (max connections): 16
+
+## Rationale
+Our experiments are designed to be fair and the same across all backends while favouring Document Stores that benefit 
+the DocArray user. Therefore, we based our benchmarks on the following:
+
+* Cover the most important operations: We understand that some backends are better at some operations than others and 
+some offer better quality. Therefore, we try to benchmark on 6 operations (CRUD + Find by vector + Find by condition)
+and report quality measurement (Recall at K).
+* Not just the speed, quality in mind: Since some backends offer better performance and some offer better quality, 
+we make sure to report the quality measurement for the Approximate Nearest Neighbor Search. This will allow users to 
+choose the backend that best suit their cases.
+* Same experiment, same API: DocArray offers the same API across all backends and therefore, we built on top of it the 
+same benchmarking experiment. Furthermore, we made sure to fix the same HNSW parameters for backends that support 
+Approximate Nearest Neighbor Search. All backends are run on official Docker containers, local to the DocArray client 
+which allows having similar network overhead. We also allocate the same resources for those Docker containers and all 
+servers are run in a single node setup.
+* Benefit the user as much as possible: We offer the same conditions and resources to all backends, but our experiment 
+favours the backends that use the resources efficiently. Therefore, some backends might not use the network, or use 
+GRPC instead of HTTP, use batch operations,... And we're okay with that, as long as it benefits the DocArray and Jina 
+user.
+* Open to improvements: We are constantly improving the performance of storage backends from the DocArray side and 
+updating the benchmarks accordingly. If you believe we missed an optimization (perform an operation in batches, benefit 
+from a recent feature in upstream, avoid unnecessary steps,...), feel free to raise a PR or issue, we're open to 
+your contributions!
+
+## Backends covered
+Our benchmarks cover all backends supported by DocArray:
+* In-memory
+* Sqlite
+* Qdrant
+* Annlite
+* Weaviate
+* ElasticSearch
```

Comment:
Add in-doc link to each, link to each's corresponding page in our docs.

Reply:
done